covalent_dispatcher/_service/app.py

```diff
@@ -97,46 +97,32 @@ def get_result(dispatch_id) -> Response:
     args = request.args
     wait = args.get("wait", default=False, type=lambda v: v.lower() == "true")
     status_only = args.get("status_only", default=False, type=lambda v: v.lower() == "true")
-    while True:
-        with Session(DispatchDB()._get_data_store().engine) as session:
-            lattice_record = (
-                session.query(Lattice).where(Lattice.dispatch_id == dispatch_id).first()
-            )
-        try:
-            if not lattice_record:
-                return (
-                    jsonify(
-                        {"message": f"The requested dispatch ID {dispatch_id} was not found."}
-                    ),
-                    404,
-                )
-            elif not wait or lattice_record.status in [
-                str(Result.COMPLETED),
-                str(Result.FAILED),
-                str(Result.CANCELLED),
-                str(Result.POSTPROCESSING_FAILED),
-                str(Result.PENDING_POSTPROCESSING),
-            ]:
-                output = {
-                    "id": dispatch_id,
-                    "status": lattice_record.status,
-                }
-                if not status_only:
-                    output["result"] = codecs.encode(
-                        pickle.dumps(result_from(lattice_record)), "base64"
-                    ).decode()
-                return jsonify(output)
-
-        except (FileNotFoundError, EOFError):
-            if wait:
-                continue
-            response = make_response(
-                jsonify(
-                    {
-                        "message": "Result not ready to read yet. Please wait for a couple of seconds."
-                    }
-                ),
-                503,
-            )
-            response.retry_after = 2
-            return response
+    with Session(DispatchDB()._get_data_store().engine) as session:
+        lattice_record = session.query(Lattice).where(Lattice.dispatch_id == dispatch_id).first()
+    if not lattice_record:
+        return (
+            jsonify({"message": f"The requested dispatch ID {dispatch_id} was not found."}),
+            404,
+        )
+    if not wait or lattice_record.status in [
+        str(Result.COMPLETED),
+        str(Result.FAILED),
+        str(Result.CANCELLED),
+        str(Result.POSTPROCESSING_FAILED),
+        str(Result.PENDING_POSTPROCESSING),
+    ]:
+        output = {
+            "id": dispatch_id,
+            "status": lattice_record.status,
+        }
+        if not status_only:
+            output["result"] = codecs.encode(
+                pickle.dumps(result_from(lattice_record)), "base64"
+            ).decode()
+        return jsonify(output)
+    response = make_response(
+        jsonify({"message": "Result not ready to read yet. Please wait for a couple of seconds."}),
+        503,
+    )
+    response.retry_after = 2

```

Comment:
Does this effectively change the behavior of `get_result(wait=True)`? Since `ct.get_result()` [retries at most 5 times](https://github.com/AgnostiqHQ/covalent/blob/7da4139384fdadca3498790417ea027202d8863d/covalent/_results_manager/results_manager.py#L169), `dispatch_sync` would no longer work with workflows that last more than 10 seconds.

Reply:
The documentation for `max_retries` says

> Note, this applies only to failed DNS lookups, socket connections and connection timeouts, never to requests where data has made it to the server.

so I think we're good?

https://requests.readthedocs.io/en/latest/api/#requests.adapters.HTTPAdapter

Reply:
Ah, I see. Thanks for the clarification